src/people.js

```diff
@@ -0,0 +1,28 @@
+class People {
+    constructor(people) {
+
+        this.name = people.name;
+        this.animals = people.animals.map(animal => animal.name);
+    }
+
+    filterAnimals(pattern) {
+        const regex = new RegExp(pattern)
+        return this.animals.filter(animal => regex.test(animal));
```

Comment:
à déplacer dans une méthode d'une nouvelle classe Animal
-->
`this.animals.filter(animal => animal.isMatching(pattern)`